yarn-project/simulator/src/public/enqueued_call_side_effect_trace.test.ts

```diff
@@ -66,56 +45,33 @@ describe('Enqueued-call Side Effect Trace', () => {
   const log = [Fr.random(), Fr.random(), Fr.random()];
   const contractInstance = SerializableContractInstance.default();
 
-  const startGasLeft = Gas.fromFields([new Fr(randomInt(10000)), new Fr(randomInt(10000))]);
-  const endGasLeft = Gas.fromFields([new Fr(randomInt(10000)), new Fr(randomInt(10000))]);
-  const transactionFee = Fr.random();
-  const calldata = [Fr.random(), Fr.random(), Fr.random(), Fr.random()];
-  const returnValues = [Fr.random(), Fr.random()];
-
-  const constants = CombinedConstantData.empty();
-  const avmEnvironment = initExecutionEnvironment({
-    address,
-    calldata,
-    transactionFee,
-  });
-  const avmCallResults = new AvmContractCallResult(/*reverted=*/ false, returnValues);
-
-  const emptyValidationRequests = PublicValidationRequests.empty();
-
   let startCounter: number;
   let startCounterFr: Fr;
   let startCounterPlus1: number;
   let trace: PublicEnqueuedCallSideEffectTrace;
 
+  const expectNonEmptyItems = <T extends IsEmpty>(array: T[], expected: T[]) => {
+    const numNonEmptyItems = countAccumulatedItems(array);
+    expect(expected.length).toBe(numNonEmptyItems);
+    expect(array.slice(0, numNonEmptyItems)).toEqual(expected);
+  };
```

Comment:
Since we are using getSideEffects now, we probably don't need to do some of this because getSideEffects returns regular TS arrays, not constant length "Tuples"

Reply:
Ohhh didn't realise that. Lovely 🙌